js/bootstrap-select.js

```diff
@@ -409,11 +399,12 @@
        * @param [optgroup]
        * @returns {string}
        */
-      var generateLI = function (content, index, classes, optgroup) {
+      var generateLI = function (content, index, classes, id, optgroup) {
```

Comment:
this is where you can add the optgroup label as a li id for the .dropdown-header